### ISSUE TITLE
Recommended valid-expect is failing if there is a message

### ISSUE THREAD

**Issue by mheinzerling** [veritem/eslint-plugin-vitest#153]
This snippet is marked as invalid by the default recommended valid-expect rule.

```
it('x', () => {
  expect('x', 'a message').toBe('y');
});
```

But it is valid and the correct syntax and behavior:

```
AssertionError: a message: expected 'x' to be 'y' // Object.is equality
Expected :y
Actual   :x
```

Following my IDE I find the type Chai.ExpectStatic

```
    export interface ExpectStatic {
        (val: any, message?: string): Assertion;
        fail(message?: string): never;
        fail(actual: any, expected: any, message?: string, operator?: Operator): never;
    }
```

So, the rule needs to be more clever, and detect the string parameter-

**Comment by veritem** [comment]
Hey, this rule triggers a warning when it's empty or parsed with more than one argument. If it's parsed with more than one argument you have to change settings for it like the following.


```json 
"vitest/valid-expect": ["error", {
    "maxArgs":  2  // <-- you can change to whatever you want. 
  }]
```

**Comment by mheinzerling** [comment]
Yes, but not every `expect` requires a message. Setting max to 2 might miss e.g.: ` expect(1,2).toBe(1)`. And setting min to 2 will require a message everywhere.

Actually I think that the parameter min/maxArgs are not suitable at all. There are only two valid option `any` and `any, string`. Do I miss anything here?

**Comment by mbtts** [comment]
@mheinzerling I do not think you are missing anything and this comes down a genuine Vitest/Jest difference in implementation.

Jest uses its own library ([expect](https://www.npmjs.com/package/expect)) and this will throw an error if it receives more than one argument ([source](https://github.com/jestjs/jest/blob/be7e797bd6d1ef3476f1d9a15fb50fd03527bd19/packages/expect/src/index.ts#L102-L105)):

```ts
export const expect: Expect = (actual: any, ...rest: Array<any>) => {
  if (rest.length !== 0) {
    throw new Error('Expect takes at most one argument.');
  }
```

I don’t know for certain, but it seems reasonable to assume that the [valid-expect](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/valid-expect.md) rule was added to [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) in order to catch this error before the tests run!

Vitest, on the other hand, is using [Chai](https://www.chaijs.com/) and implementing the whole Jest expect API as a chai plugin ([source](https://github.com/vitest-dev/vitest/blob/ba1e7335f93b82c110209a6b6e3099908aa1b2f8/packages/vitest/src/integrations/chai/setup.ts#L7)):


```ts
chai.use(JestChaiExpect);
```

In the case of Jest it makes sense for the `minArgs` to default to one because, although libraries like [jest-expect-message](https://www.npmjs.com/package/jest-expect-message) _can_ add support for an additional second message parameter, one parameter is the default.

Whereas Chai will always accept a second parameter for a message. So for as long as Vitest uses Chai for its matchers it will inherit this behaviour. Thus it makes sense to me for two arguments to be the default for this rule in eslint-plugin-vitest, but **not** in eslint-plugin-jest.

Two arguments would also make this rule consistent with the [type definition](https://github.com/vitest-dev/vitest/blob/ba1e7335f93b82c110209a6b6e3099908aa1b2f8/packages/expect/src/types.ts#L81-L93) of expect provided by Vitest:

```ts
export interface ExpectStatic extends Chai.ExpectStatic, AsymmetricMatchersContaining {
  <T>(actual: T, message?: string): Assertion<T>
  // …
}
```
